src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp

```diff
@@ -129,6 +129,31 @@ setupLuks( const LuksDevice& d )
     return true;
 }
 
+// static
+QVariantList
+LuksBootKeyFileJob::partitions()
+{
+    Calamares::GlobalStorage* globalStorage = Calamares::JobQueue::instance()->globalStorage();
+    return globalStorage->value( QStringLiteral( "partitions" ) ).toList();
+}
```

Comment:
Congratulations on writing your first C++! Now I see where your earlier confusion about `static` came from. This function does not use any of the class members; it doesn't do anything that is special to the class. So you can remove the `LuksBootKeyFileJob::` there: that makes it a "free function" not associated with any class. Then you remove the declaration of the function from the header, too.

The thing with free functions is that they're "visible" on the outside, and sometimes they are not interesting, useful or whatever for the outside. In order to hide them, C++ has a variety of things you can use. In Calamares, we write `static` in front of free functions that should be local to the C++ file they are defined in.

Reply:
Great! Not knowing C++, I did not know that you can write free functions, but that is definitely preferable here.

Reply:
In my defense (since I wrote the code from `dracutlukscfg` on which you based this, using private static member functions), I would like to point out that several C++ developers consider file-scope functions (free functions declared as `static`) deprecated, and that using private static member functions is the object-oriented approach. On the other hand, the static member functions do pollute the header file with private functions that do not need to be there, so I am not opposed to using free functions instead. We can change `dracutlukscfg` as well. Or we could even make the functions truly global rather than file-scope, to avoid the code duplication (but then they would have to be somewhere in `libcalamares`, I guess).

Reply:
Sharing them globally and running only once would probably the cleanest way. But it is a very small check, so it doesn't probably matter that much.